### PR TITLE
Initial "2.0 upgrade guide" in docs

### LINKED ISSUE #3434: "2.0 upgrade guide" in docs
Migrated from GitLab: https://gitlab.com/meltano/meltano/-/issues/3519

Originally created by @aaronsteers on 2022-05-22 01:33:21

---

We should create a docs page with instructions on how to migrate to 2.0.

### PR DESCRIPTION
**Pull request by MeltyBot**
Migrated from GitLab: https://gitlab.com/meltano/meltano/-/merge_requests/2660

Originally created by @aaronsteers on 2022-05-22 01:33:47

---

## Merge Request Checklist

- [ ] Link to an issue
- [ ] Include the proposed fix or feature
- [ ] Include and update tests for the modified code
- [ ] Include a documentation change
- [ ] Add a [CHANGELOG.md](https://gitlab.com/meltano/meltano/blob/master/README.md#Changelog) entry in the Unreleased section for any user-facing changes
- [ ] Mention or assign a maintainer

Closes https://github.com/meltano/meltano/issues/3434

### CODE REVIEW

File: docs/src/_reference/v2-migration.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+1. **Removed: `model` plugin type**
+   - This plugin type provided very basic BI capabilities using Meltano UI. This functionality has been removed in favor of 3rd party open source BI solutions.
+1. **Removed: `dashboard` plugin type**
+   - This plugin type provided very basic BI capabilities using Meltano UI. This functionality has been removed in favor of 3rd party open source BI solutions.
+1. **Removed: `env_aliases` in plugin config**
+1. **Removed: transforms support in `meltano elt`**
+   - Meltano 2.0 continues to support extract-load (EL) operations with `meltano elt`. However, for EL+T operations which also need to transform data, please use `meltano run`.
+1. **Removed: transforms support in Meltano schedules**
+   - Meltano 2.0 continues to support extract-load (EL) operations in schedules. However, for EL+T operations which also need to transform data, please use the new `meltano job add` command to create a job definition and then specify the new job name in your schedule.
+1. **Recommended: migrating to an adapter-specific dbt plugin
```
Comment (aaronsteers) [inline docs/src/_reference/v2-migration.md:23]:
```suggestion
1. **Recommended: migrating to an adapter-specific dbt plugin**
```

File: docs/src/_reference/v2-migration.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+---
+title: Meltano 2.0 Migration Guide
+description: Migrate existing "v1" projects to the latest version of Meltano
+layout: doc
+redirect_from:
+  - /guide/ui/
+weight: 10
+---
+
```
Comment (aaronsteers) [inline docs/src/_reference/v2-migration.md:9]:
```suggestion

_Note: This document is still a work in progress. Expect further changes, coming soon._

```

**Comment by aaronsteers** [comment]
@afolson and @tayloramurphy - I'd like to go ahead and merge those so that other can continue to expand the list from here.

**Comment by afolson** [comment]
@aaronsteers Thanks for this! I don't think this was published yet because of the repo migration 😬 

@pandemicsyn It looks like I can just change the repo this deploys from and keep the same settings. Do you know offhand if anything will explode? I'm willing to do science.

**Comment by pandemicsyn** [comment]
@afolson send it! Should just work™

**Pull request merged**